os/osx.cmake

```diff
@@ -58,6 +58,21 @@ set(
     CACHE STRING "System root for OSX" FORCE
 )
 
+# With a full Xcode install, typically `xcode-select -print-path` is something like:
+#     /Applications/Xcode.app/Contents/Developer
+#
+# But with just Xcode command line tools installed, the path is:
+#     /Library/Developer/CommandLineTools
+#
+# In the CommandLineTools case, the SDKs folder is at a different relative path.
+if(NOT EXISTS "${CMAKE_OSX_SYSROOT}")
+  set(
+      CMAKE_OSX_SYSROOT
+      "${XCODE_DEVELOPER_ROOT}/SDKs/MacOSX${OSX_SDK_VERSION}.sdk"
+      CACHE STRING "System root for OSX" FORCE
+  )
+endif()
+
 if(NOT EXISTS "${CMAKE_OSX_SYSROOT}")
   polly_fatal_error("${CMAKE_OSX_SYSROOT} not exists")
```

Comment:
Now this error will not be accurate. If SDK is not installed it will emit error message "/Library/Developer/CommandLineTools/SDKs/MacOSXa.b.sdk not exists" however in most cases it should be "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSXa.b.sdk not exists".

I think it's better to do:
```cmake
set(__osx_sysroot_suggestion_1 "${XCODE_DEVELOPER_ROOT}/Platforms/MacOSX.platform/Developer/SDKs/MacOSX${OSX_SDK_VERSION}.sdk")
set(__osx_sysroot_suggestion_2 "${XCODE_DEVELOPER_ROOT}/SDKs/MacOSX${OSX_SDK_VERSION}.sdk")

if(EXISTS "${__osx_sysroot_suggestion_1}")
  set(__osx_sysroot "${__osx_sysroot_suggestion_1}")
elseif(EXISTS "${__osx_sysroot_suggestion_2}")
  set(__osx_sysroot "${__osx_sysroot_suggestion_2}")
else()
  # error message with both __osx_sysroot_suggestion_1 and __osx_sysroot_suggestion_2
endif()

# set CMAKE_OSX_SYSROOT to __osx_sysroot
```

Reply:
Thanks for the suggestion, done.